### DISCUSSION TITLE
FFD: Full fluid volume distorsion when running optimization

### DISCUSSION THREAD

**Discussion by AlbertoVilariniTarrio** [mdolab/dafoam#637] [General]
Hello everybody!

My name is Alberto, I am a PhD student at Carlos III de Madrid University. My investigation fields are automotive aerodynamics and machine learning. For the last month I have been working with DAFoam, I review the available documentation, run some of the tutorial cases, view the three available webminars and used the forum to solve some doubts and problems I faced.

As part of my learning process, I created a simple case that I wanted to resolve using DAFoam: a quasi 2D cylinder (it is a 3D case, but the cylinder pass through the control volume) at very low speed for computational reasons. The coordinate system used is x as longitudinal direction (positive following the flow freestream), y as transversal direction and z as vertical direction (upwards as positive).

I modify the OpenFOAM directories as required to implement the modifications needed to simulate the cylinder under the conditions chosen. As well, I changed the runScript accordingly to my needs (for example, remove the angle of attack, change the constraints and design variables, etc) and modify the genFFD.py script to generate a the FFD points to my geometry.

The design surface is the cylinder, and the FFD points generated for this potimization problem are illustrated in the images below.

Once all of this was done, I run the case and get some results, where the geometry changes evolve as expected. However, all the volumetric mesh is deformed, I was expecting the deformation just to be applied to the cylinder and volume mesh inside my FFD volume, despite of this all my volume is modified, reaching a situation where the volume mesh twists (see images below).

Is there any way to prevent this to happen? Can I define what areas of the volume mesh are morphed and which ones stay as they are? I used adjointOptimizatoinFoam and in that case it is offered the possibility to confine certain FFD points in order to use them as transition between the distorted and undistorted mesh.

Attache the run script as well.

Thanks for your time and attention, hope you can give me some help.

Regards,

Alberto

![FFD_points](https://github.com/mdolab/dafoam/assets/169068787/f6a2d403-733d-42f9-a1a8-b943ebd96594)
![FFD_points_full_volume](https://github.com/mdolab/dafoam/assets/169068787/356a0996-5a3a-4c41-8e11-f42140758b32)
![Cylinder 0009](https://github.com/mdolab/dafoam/assets/169068787/efe94951-e98d-4c4b-bf65-831bf6059852)
![Cylinder 0013](https://github.com/mdolab/dafoam/assets/169068787/9a9efda9-0e6c-4528-8e85-28f98d0f8edc)
![Cylinder 0015](https://github.com/mdolab/dafoam/assets/169068787/13e38779-f1e2-4708-b861-9227e335acc8)
![Cylinder 0017](https://github.com/mdolab/dafoam/assets/169068787/3383c8c0-f2d0-41c6-ae
![Cylinder 0020](https://github.com/mdolab/dafoam/assets/169068787/91d61eab-3ffd-4d71-9e9f-48c76083fb34)
5e-8ec16ca4c173)
![Cylinder 0021](https://github.com/mdolab/dafoam/assets/169068787/d1aa70a8-af4
![Cylinder 0022](https://github.com/mdolab/dafoam/assets/169068787/176ca694-7268-45e5-9dd4-9826ecbbc61c)
d-4eef-9bbd-1ea6c0d51abf)
![Cylinder 0023](https://github.com/mdolab/dafoam/assets/169068787/3aa513e2-1e6c-4fa6-ae0c-12d21d673a66)
[runScript.txt](https://github.com/mdolab/dafoam/files/15222509/runScript.txt)

**Comment by friedenhe** [comment]
I suspect the problem is in your BC setup. Your case is exactly the same as the NACA0012/incompressible tutorial, except that your geometry is a cylinder instead of an airfoil. I suggest you start from the airfoil case and gradually change it to your setup, one step at a time. Then, you should be able to figure out why part breaks the code. I suggest you start with simply replacing the airfoil geometry with the cylinder without changing anything in the BC setup and runScript.py. 

On a separate note, you don't need that many FFD points in the y direction.
